### PR TITLE
boards: arm: Board Creation for cy8cproto_063_ble 

### PR DESCRIPTION
**Pull request by sreeramIfx**
Adding support for Infineon cy8cproto_063_ble board

### CODE REVIEW

**Comment by carlescufi** [comment]
@npal-cy

**Comment by npal-cy** [comment]
Hi @sreeramIfx, you need to add dts\soc support for PSoC6 1M

DTS part:
dts\arm\infineon\psoc6\mpns\CYBLE_416045_02.dtsi (you can add all  PSoC6 1M MPNs)
dts\arm\infineon\psoc6\psoc6_01\*.*

SOC part:
\soc\arm\infineon_cat1\psoc6\Kconfig.defconfig.soc.psoc6_01
\soc\arm\infineon_cat1\psoc6\Kconfig.soc.psoc6_01

File: boards/arm/cy8cproto_063_ble/cy8cproto_063_ble.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+toolchain:
+  - zephyr
+  - gnuarmemb
+supported:
+  - bluetooth
+  - counter
+  - gpio
+  - uart
+  - i2c
+  - pwm output
```
Comment (npal-cy) [inline boards/arm/cy8cproto_063_ble/cy8cproto_063_ble.yaml:22]: let's enable supported feature once it added to mainline, so for:
```
supported:
  - gpio
  - uart
```

File: boards/arm/cy8cproto_063_ble/Kconfig.defconfig [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# CY8CPROTO-062-4343W PSoC™ 6 Wi-Fi BT Prototyping Kit configuration
```
Comment (npal-cy) [inline boards/arm/cy8cproto_063_ble/Kconfig.defconfig:1]: update board name "CY8CPROTO-063-BLE PSoC™ 6 BLE Prototyping Kit"

File: boards/arm/cy8cproto_063_ble/cy8cproto_063_ble_defconfig [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+CONFIG_SERIAL=y
+
+# Enable pin controller
+CONFIG_PINCTRL=y
+
+# Enable clock controller
+CONFIG_CLOCK_CONTROL=y
+
+# Enable assert
+CONFIG_ASSERT=y
```
Comment (npal-cy) [inline boards/arm/cy8cproto_063_ble/cy8cproto_063_ble_defconfig:32]: remove this CONFIG_ASSERT.

File: boards/arm/cy8cproto_063_ble/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+   west flash --openocd path/to/infineon/openocd/bin/openocd
+   west debug --openocd path/to/infineon/openocd/bin/openocd
+
+References
+**********
+
+.. _PSoC 63 BLE MCU SoC Website: https://www.infineon.com/cms/en/product/evaluation-boards/cy8cproto-063-ble/
+
+.. _PSoC 63 BLE MCU Datasheet:
```
Comment (npal-cy) [inline boards/arm/cy8cproto_063_ble/doc/index.rst:106]: missed links to web resources...

**Review by npal-cy — DISMISSED** [review]

**Review by npal-cy — DISMISSED** [review]

**Review by ifyall — DISMISSED** [review]

**Review by galak — CHANGES_REQUESTED** [review]
Can you reduce the duplication in the CY*.dtsi files.  Not 100% sure what the differences all are, but seems alot of them are just about mem/flash sizes.  In those cases, they dtsi should just update the reg property for those nodes.

**Review by npal-cy — DISMISSED** [review]

File: dts/arm/infineon/psoc6/mpns/CY8C6336BZI_BLD14.dtsi [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+ */
+
+#include <arm/armv7-m.dtsi>
+#include "../psoc6_01/psoc6_01.124-bga-sip.dtsi"
+
+
+/ {
+
+
+
```
Comment (galak) [inline dts/arm/infineon/psoc6/mpns/CY8C6336BZI_BLD14.dtsi:15]: can you cleanup the whitespace here (remove blank lines), and in other similar files.

**Review by galak — CHANGES_REQUESTED** [review]
Can you refactor the duplication on the dts/arm/infineon/psoc6/psoc6_01/psoc6*.dtsi files.  Looks like a fair amount of duplication under the pinctrl node.

File: dts/arm/infineon/psoc6/mpns/CY8C6336BZI_BLD14.dtsi (hunk 5 above)
Comment (sreeramIfx) [reply dts/arm/infineon/psoc6/mpns/CY8C6336BZI_BLD14.dtsi]: updated..

**Comment by npal-cy** [comment]
Hi @galak 

> Can you refactor the duplication on the dts/arm/infineon/psoc6/psoc6_01/psoc6*.dtsi files. Looks like a fair amount of duplication under the pinctrl node.

package dtsi (dts/arm/infineon/psoc6/psoc6_01/psoc6_01.xxxxx.dtsi) reflect gpio resources and pinctrl nodes which is suitable for this  packages. Example if device has psoc6_01.124-bga-sip.dtsi we declare pinctrl for uart_scb0...uart_scb7, if it  psoc6_01.43-smt.dtsi we uart_scb0...uart_scb6 (we do not add uart_scb7, uart_scb8 because in does not supported by this kind of package). Let me know if it makes sense.

**Review by npal-cy — DISMISSED** [review]

**Comment by ifyall** [comment]
@galak, do you have any more questions on the DTSI files here?

**Review by ifyall — DISMISSED** [review]

**Review by galak — CHANGES_REQUESTED** [review]
Some of the shared nodes in in the psoc6_01.xxx*dtsi files for things like hsiom, gpio, etc should be in a common dtsi file.

**Review by galak — CHANGES_REQUESTED** [review]
The uart/i2c/spi/... as a scb node and than in the board dts it modifies the compatible for i2c/spi/uart.

Look at ATMEL sercom or NXP flecomm to see examples:
https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/adafruit_feather_m0_lora/adafruit_feather_m0_lora.dts#L43

Here's a rough example of what the dtsi:

```
		scb4: scb@40650000 {
			compatible = "infineon,cat1-scb";
			reg = <0x40650000 0x10000>;
			#address-cells = <1>;
			#size-cells = <0>;
			interrupts = <45 6>;
			status = "disabled";
		};
		
		scb5: scb@40660000 {
			compatible = "infineon,cat1-scb";
			reg = <0x40660000 0x10000>;
			#address-cells = <1>;
			#size-cells = <0>;
			interrupts = <46 6>;
			status = "disabled";
		};	
```

**Comment by sreeramIfx** [comment]
> The uart/i2c/spi/... as a scb node and than in the board dts it modifies the compatible for i2c/spi/uart.
> 
> Look at ATMEL sercom or NXP flecomm to see examples: https://github.com/zephyrproject-rtos/zephyr/blob/main/boards/arm/adafruit_feather_m0_lora/adafruit_feather_m0_lora.dts#L43
> 
> Here's a rough example of what the dtsi:
> 
> ```
> 		scb4: scb@40650000 {
> 			compatible = "infineon,cat1-scb";
> 			reg = <0x40650000 0x10000>;
> 			#address-cells = <1>;
> 			#size-cells = <0>;
> 			interrupts = <45 6>;
> 			status = "disabled";
> 		};
> 		
> 		scb5: scb@40660000 {
> 			compatible = "infineon,cat1-scb";
> 			reg = <0x40660000 0x10000>;
> 			#address-cells = <1>;
> 			#size-cells = <0>;
> 			interrupts = <46 6>;
> 			status = "disabled";
> 		};	
> ```

@galak we will have to make these changes to the already merged [psoc6_02.dtsi](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/infineon/psoc6/psoc6_02/psoc6_02.dtsi) as well. Can we make these changes in a separate PR? We want to review the implementations of ATMEL/NXP and work on our implementation as a separate task.

**Comment by sreeramIfx** [comment]
> Some of the shared nodes in in the psoc6_01.xxx*dtsi files for things like hsiom, gpio, etc should be in a common dtsi file.

moved the gpio, hsiom nodes to a common file and used /delete-node/ to configure per package..

**Comment by galak** [comment]
> @galak we will have to make these changes to the already merged [psoc6_02.dtsi](https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/arm/infineon/psoc6/psoc6_02/psoc6_02.dtsi) as well. Can we make these changes in a separate PR? We want to review the implementations of ATMEL/NXP and work on our implementation as a separate task.

We should figure this out now.  The impact to psoc6_02.dtsi seems pretty small at this point since it only seems to be defining uart's and not all the other I2C, SPI, etc. nodes.

**Review by galak — APPROVED** [review]
LGTM.  Thanks.

We should follow up with updating the i2c, uart, etc bindings to include the `infineon,cat1-scb.yaml` to convey the relationship.

**Review by npal-cy — APPROVED** [review]

File: boards/arm/cy8cproto_063_ble/doc/index.rst [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
++-----------+------------+-----------------------+
+| Counter   | on-chip    | Counter               |
++-----------+------------+-----------------------+
+| Bluetooth | on-chip    | Bluetooth             |
++-----------+------------+-----------------------+
+
+
+The default configurations can be found in the Kconfig
+
+:zephyr_file:`boards/arm/cy8cproto-063-ble/cy8cproto_063_defconfig`
```
[the remark this answers is not in the document]
Comment (sreeramIfx) [reply boards/arm/cy8cproto_063_ble/doc/index.rst:63]: updated @ https://github.com/zephyrproject-rtos/zephyr/pull/57814

**Comment by sreeramIfx** [comment]
> 

Updated @ https://github.com/zephyrproject-rtos/zephyr/pull/57814

**Pull request merged**